modules/openapi-generator/src/main/resources/csharp-netcore/ApiClient.mustache

```diff
@@ -408,7 +410,26 @@ namespace {{packageName}}.Client
 
             InterceptRequest(req);
 
-            var response = client.Execute<T>(req);
+            IRestResponse<T> response;
+            if (configuration.RetryStatusCodes != null)	
+            {
+                Random jitterer = new Random();
+                var policy = Policy
+                    .HandleResult<IRestResponse<T>>(resp => (configuration.RetryStatusCodes.Contains(resp.StatusCode)))
+                    .WaitAndRetry(configuration.MaxRetries, retryAttempt => TimeSpan.FromSeconds(Math.Pow(2, retryAttempt))
+                                                          + TimeSpan.FromMilliseconds(jitterer.Next(0, 1000))
+                     );
+                var policyResult = policy.ExecuteAndCapture(() => client.Execute<T>(req));
+                response = (policyResult.Outcome == OutcomeType.Successful) ? policyResult.Result : new RestResponse<T>
+                {
+                    Request = req,
+                    ErrorException = policyResult.FinalException
+                };
+            }
+            else
+            {
+                response = client.Execute<T>(req);
+            }
```

Comment:
The inability to customize the retry strategy is a concern, and I think a design that injects the retry strategy via IConfiguration would be better.

Reply:
@shibayan agree with you, thought of implementing that way itself, but the return type being a generic type IRestReponse<T> is what making it difficult to implement it via configuration.
Or least i couldn't think of a solution to implement that in a easier way. Any inputs along that lines will be helpful.

Reply:
@shibayan  Figured out a way to have the retry policy through configuration. As a workaround found that the RestSharp's ExecAsync< T > can be seperated in two steps ExecAsync and then Deserialize< T > which makes ExecAsync free from having generic type and thus i can define the retry policy free from generic and so able to inject through configuration..
Please take a look at it once...

Reply:
@shibayan can you take a look again when you get time.